tests/resources/test_generative_design_execution.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+import uuid
+
+from citrine.informatics.generative_design import GenerativeDesignInput
+from citrine.informatics.executions.generative_design_execution import GenerativeDesignExecution
+from citrine.resources.generative_design_execution import GenerativeDesignExecutionCollection
+from citrine.informatics.generative_design import FingerprintType
+from tests.utils.session import FakeSession, FakeCall
+
+
+@pytest.fixture
+def session() -> FakeSession:
+    return FakeSession()
+
+
+@pytest.fixture
+def collection(session) -> GenerativeDesignExecutionCollection:
+    return GenerativeDesignExecutionCollection(
+        project_id=uuid.uuid4(),
+        session=session,
+    )
+
+
+@pytest.fixture
+def generative_design_execution(collection: GenerativeDesignExecutionCollection, generative_design_execution_dict) -> GenerativeDesignExecution:
+    return collection.build(generative_design_execution_dict)
+
+
+def test_basic_methods(generative_design_execution, collection):
+    assert "GenerativeDesignExecution" in str(generative_design_execution)
+
+    with pytest.raises(NotImplementedError):
+        collection.register(generative_design_execution)
+
+    with pytest.raises(NotImplementedError):
+        collection.update(generative_design_execution)
+
+
+def test_build_new_execution(collection, generative_design_execution_dict):
+    # Given
+    generative_design_execution_id = uuid.uuid4()
+    build_data = generative_design_execution_dict.copy()
+    build_data["id"] = str(generative_design_execution_id)
+
+    # When
+    execution: GenerativeDesignExecution = collection.build(build_data)
+
+    # Then
+    assert execution.uid == generative_design_execution_id
+    assert execution.project_id == collection.project_id
+    assert execution._session == collection.session
+    assert execution.in_progress() and not execution.succeeded() and not execution.failed()
```

Comment:
Why generate a new uid to assign to `build_data`? Why not just take the uid that comes with `generative_design_execution_dict` and assert that that uid makes it on to the `GenerativeDesignExecution` object?

Reply:
For example, the following test passes and I believe tests the same thing:
```
    execution: GenerativeDesignExecution = collection.build(generative_design_execution_dict)

    assert str(execution.uid) == generative_design_execution_dict["id"]
    assert execution.project_id == collection.project_id
    assert execution._session == collection.session
    assert execution.in_progress() and not execution.succeeded() and not execution.failed()
```

Reply:
I was trying to use the pattern found here. I'm not sure if there is a "gotcha" they were avoiding that we don't need to worry about when that workflow_id is included. I happy making it simpler by removing the `generative_design_execution_id = uuid.uuid4()` though.

https://github.com/CitrineInformatics/citrine-python/blob/main/tests/resources/test_predictor_evaluation_executions.py#L47